src/impl_x86__base_implementation.inl

```diff
@@ -1634,6 +1641,10 @@ CacheInfo GetX86CacheInfo(void) {
     if (IsBitSet(leaves.leaf_80000001.ecx, 22)) {
       ParseCacheInfo(leaves.max_cpuid_leaf_ext, 0x8000001D, &info);
     }
+  } else if (IsVendor(leaves.leaf_0, CPU_FEATURES_VENDOR_ZHAOXIN_CTH) || 
+	           IsVendor(leaves.leaf_0, CPU_FEATURES_VENDOR_ZHAOXIN_SH)) {
+    ParseLeaf2(&leaves, &info);
+    ParseCacheInfo(leaves.max_cpuid_leaf, 4, &info); 
```

Comment:
duplicated code as intel, could you combine this logic?

Reply:
Yes，I have combined the same logic in GetX86CacheInfo.